File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        db1 = {}
        db2 = {}

        for char in s:
            if char not in db1:
                db1[char] = 0
            db1[char] += 1
        
        for char in t:
            if char not in db2:
                db2[char] = 0
            db2[char] += 1
        
        if len(db1) != len(db2):
            return False

        else:

            for item in db1:
                if item not in db2:
                    return False
                else:
                    if db1[item] != db2[item]:
                        return False
        
        return True

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int longest = 0;
        ArrayList<Character> db = new ArrayList<>();
        char[] charArray = s.toCharArray();
        for(char item : charArray){
            if(!db.contains(item)){
                db.add(item);
            }
            else{
                if(db.get(0) == item){
                    db.remove(0);
                }
                else{
                    if(db.size() > longest){
                        longest = db.size();
                    }
                    db.clear();
                }
                db.add(item);
                
            }
        }
        if(db.size() > longest){
            longest = db.size();
        }

        return longest;
        

        
    }
}
